pkg/resourcemanager/controller/tokenrequestor/reconciler_test.go

```
@@ -0,0 +1,300 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokenrequestor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1fake "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/client-go/testing"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Reconciler", func() {
+	Describe("#Reconcile", func() {
+		var (
+			ctx = context.TODO()
+
+			logger     logr.Logger
+			fakeClock  clock.Clock
+			fakeJitter func(time.Duration, float64) time.Duration
+
+			sourceClient, targetClient client.Client
+			coreV1Client               *corev1fake.FakeCoreV1
+
+			ctrl *reconciler
+
+			secret         *corev1.Secret
+			serviceAccount *corev1.ServiceAccount
+			request        reconcile.Request
+
+			secretName              string
+			serviceAccountName      string
+			serviceAccountNamespace string
+			expectedRenewDuration   time.Duration
+			token                   string
+			fakeNow                 time.Time
+
+			fakeCreateServiceAccountToken = func() {
+				coreV1Client.AddReactor("create", "serviceaccounts", func(action testing.Action) (bool, runtime.Object, error) {
+					if action.GetSubresource() != "token" {
+						return false, nil, fmt.Errorf("subresource should be 'token'")
+					}
+
+					cAction, _ := action.(testing.CreateAction)
+					tr, _ := cAction.GetObject().(*authenticationv1.TokenRequest)
```

It's not good practice to ignore errors and type conversion indicators in tests.